File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/exception/RoleManagementExceptionFactory.java
package com.nivasafinance.features.rolemanagement.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import org.springframework.context.MessageSource;

public final class RoleManagementExceptionFactory {
    
    private RoleManagementExceptionFactory() {
    }
    
    public static RoleManagementNotFoundException notFound(String entityKey, Long id, MessageSource messageSource) {
        String messageKey = "error.rolemanagement." + entityKey + ".not.found";
        String message = ExceptionUtils.createLocalizedMessage(messageKey, new Object[]{id.toString()}, messageSource);
        return new RoleManagementNotFoundException(message);
    }
}

File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/exception/RoleManagementExceptionFactory.kt
package com.nivasafinance.features.rolemanagement.exception

import com.nivasafinance.common.exception.ExceptionUtils
import org.springframework.context.MessageSource
import java.util.UUID

object RoleManagementExceptionFactory {

    fun notFound(entityKey: String, id: UUID, messageSource: MessageSource): RoleManagementNotFoundException {
        val messageKey = "error.rolemanagement.$entityKey.not.found"
        val message = ExceptionUtils.createLocalizedMessage(messageKey, arrayOf(id.toString()), messageSource)
        return RoleManagementNotFoundException(message)
    }
}

class RoleManagementNotFoundException(message: String) : RuntimeException(message)


